### ISSUE TITLE
No Inner Declarations Rule

### ISSUE THREAD

**Issue by jigawot** [buzinas/tslint-eslint-rules#97]
As far as I can tell, the rule about declaring nested functions is as follows:

In strict mode, function declarations cannot be nested inside a statement or block. They may only appear at the top level or directly inside a function body.

Thus, if you declare

```TypeScript
namespace MyNamespace {
  function MyHelperFunction() {
    console.log("This _should_ be allowed.");
  }
}
```

it will be compiled into something that looks vaguely like this (depending on your module output setting)

```JavaScript
var MyNamespace;
(function (MyNamespace) {
  function MyHelperFunction() {
    console.log("This _should_ be allowed.");
  }
})(MyNamespace || (MyNamespace = {}));
```

This, therefore, does not violate the rule about nested functions, but the linter still marks it as invalid.  I propose we add `ts.SyntaxKind.ModuleDeclaration` to the list of valid parent types in noInnerDeclarationsRule.ts and the appropriate associated test case.

I have a fork and can produce a PR if approved.

**Comment by buzinas** [comment]
I would be glad to receive a PR for this.

**Comment by jmlopez-rod** [comment]
This issue can now be closed since the PR has been merged: https://github.com/buzinas/tslint-eslint-rules/pull/98
